cowprotocol/cowswap
# 332/Store slippage info on appData

# Summary

Part of #332 

## Edit 2022-07-01

No longer creating on appData per order.

Now stores the selected slippage, which will significantly reduce the amount of appData uploaded, while still capturing the original price.

---

## Original description

Creating a new appDataHash for every order.

It's recalculated for every price quote.
It includes:
- quote metadata (buy/sell amounts)
- referrer metadata - if referral address is set and is valid
- appCode - CowSwap or CowSwap-SafeApp
- environemnt - local, dev, prod, barn, ens, etc

  # To Test

1. Load the page and open the dev console.
2. Filter by `appData`
3. On every price change you should see a debug log similar to this image
<img width="491" alt="Screen Shot 2022-06-08 at 15 03 49" src="https://user-images.githubusercontent.com/43217/172636812-6a89b329-07b3-47a0-b5a2-006c17e7bcdb.png">

* Make sure the fields match what you are seeing:
* environment: `pr`
* CID and appDataHash: calculated but won't exist yet on IPFS
4. Place an order and open it on the browser `https://barn.api.cow.fi/mainnet/api/v1/orders/<orderId>`
* Check the appDataHash in the order matches the last console debug log
5. Filter the console log for `UploadToIpfs`
* Check the appData related to your order has been uploaded, like the image
<img width="475" alt="Screen Shot 2022-06-08 at 15 10 57" src="https://user-images.githubusercontent.com/43217/172638536-5634007f-d9fe-4a76-b77c-c624bec0d6ba.png">

7. Search for the appDataHash using https://codepen.io/pacara/pen/BadZgjg?editors=0101
* Check it exists and the fields match the last console debug log
